[IMP] l10n_br: New fiscal regime update + new loc template (#18377)
odoo/documentation: content/applications/finance/fiscal_localizations/brazil.rst

   * - :guilabel:`Brazil - Accounting Reports`
     - `l10n_br_reports`
     - Accounting reports for Brazil.
   * - :guilabel:`Brazil Pix QR codes`
     - `l10n_br_pix`
     - Implements Pix QR codes for Brazil.
   * - :guilabel:`Brazil - Sale Subscription`
     - `l10n_br_sale_subscription`

>      - Sale subscription modifications for Brazil.

Reviewer: ```suggestion
     - Subscriptions modifications for Brazil.
```
I would use the app name (Subscriptions) even if the module name doesn't follow that naming convention.

   * - :guilabel:`Brazil - Sale`
     - `l10n_br_sales`
     - Sales modifications for Brazil.
   * - :guilabel:`Brazil - Website Sale`
     - `l10n_br_website_sale`
     - Allows tax calculation and EDI for eCommerce users.

Additionally, the following modules must be manually :ref:`installed <general/install>`:

.. list-table::
   :header-rows: 1
   :widths: 25 30 50

   * - Name
     - Technical name
     - Description

>    * - :guilabel:`AvaTax Brazil` & :guilabel:`AvaTax Brazil for Services` & :guilabel:`Test SOs for

Reviewer: ```suggestion
   * - :guilabel:`Avatax Brazil` & :guilabel:`Avatax Brazil for Services` & :guilabel:`Test SOs for
```
That's how they are spelled in Odoo (unfortunately ;))

       the Brazilian Avatax`
     - `l10n_br_avatax` & `l10n_br_avatax_services` & `l10n_br_test_avatax_sale`

>      - Goods and Services tax computation through Avalara.

Reviewer: ```suggestion
     - Goods and services tax computation through Avalara.
```

   * - :guilabel:`Avatax Brazil eCommerce Fiscal Reform`
     - `l10n_br_website_sale_fiscal_reform`
     - eCommerce modifications for fiscal reform.
   * - :guilabel:`Avatax Brazil Fiscal Reform`
     - `l10n_br_edi_fiscal_reform`
     - Avatax modifications for fiscal reform.
   * - :guilabel:`Avatax Brazil Sale Fiscal Reform`
     - `l10n_br_edi_sale_fiscal_reform`
     - Sales Avatax modifications for fiscal reform
   * - :guilabel:`Avatax Brazil Sale for Services`
     - `l10n_br_edi_sale_services`
     - Sales modifications for Brazil for services
   * - :guilabel:`Brazilian Accounting EDI` & :guilabel:`Brazilian Accounting EDI for services`
     - `l10n_br_edi` & `l10n_br_edi_services`
     - Provides electronic invoicing for goods and services for Brazil through AvaTax.
   * - :guilabel:`Brazilian Accounting EDI For Sale`
     - `l10n_br_edi_sale`

>      - Adds some fields to sale orders that will be carried over the invoice.

Reviewer: ```suggestion
     - Adds fields to sales orders that will be carried over to the invoice.
```

  good or service is used in the transaction, as this can affect tax calculation and invoice
  issuance.

If the selected :guilabel:`Product Type` is :guilabel:`Service`, select the :guilabel:`NBS Code`
(Nomenclatura Brasileira de Serviços) necessary for tax classification.

.. _localizations/brazil/company-and-contacts:

Company and contacts
====================

To use all the features of this fiscal localization, the following fields are required on the
:doc:`company record </applications/general/companies>`:

- :guilabel:`Name`
- :guilabel:`Address`: add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
  :guilabel:`Country`

  - In the :guilabel:`Street` field, enter the street name, number, and any additional address
    information.
  - In the :guilabel:`Street 2` field, enter the neighborhood.

- :guilabel:`Identification Number`: :guilabel:`CNPJ` or :guilabel:`CPF`
- :guilabel:`Tax ID`: associated with the identification type
- :guilabel:`IE`: State registration
- :guilabel:`IM`: Municipal registration
- :guilabel:`SUFRAMA code`: Superintendence of the Manaus Free Trade Zone - add if applicable
- :guilabel:`Phone`
- :guilabel:`Email`

> Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales and Purchase` tab:

Reviewer: ```suggestion
Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales & Purchase` tab:
```
You won't find this tab under Settings > Companies > Manage Companies (as per the "company record" ref/link). As you know, you need to go through the Contacts app.
Unless this info isn't related to the DB's company, but to customers' companies? I would make it clearer what is relevant to your company, and what is relevant for customer companies.

   - Add the :ref:`Fiscal Position <localizations/brazil/fiscal-positions>` for AvaTax Brazil.

>    - :guilabel:`Tax Regime`: Federal Tax Regime.

Reviewer: ```suggestion
   - :guilabel:`Tax Regime`: Federal tax regime.
```

   - :guilabel:`ICMS Taxpayer Type`: Select :guilabel:`ICMS regime`, :guilabel:`Exempt status`,
     or :guilabel:`Non-Taxpayer`.
   - :guilabel:`Main Activity Sector`

Configure the following extra :guilabel:`Fiscal Information` to issue NFS-e:

   - Add the :ref:`Fiscal Position <localizations/brazil/fiscal-positions>` for AvaTax Brazil.
   - :guilabel:`COFINS Details`: :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt,
     Suspended`
   - :guilabel:`PIS Details` :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt,

>      Suspended`

Reviewer: Each element should be a separate guilabel.
```suggestion
- :guilabel:`COFINS Details`: :guilabel:`Taxable`, :guilabel:`Not Taxable`, :guilabel:`Taxable With Rate=0.00`, :guilabel:`Exempt`,
    :guilabel:`Suspended`
- :guilabel:`PIS Details`: :guilabel:`Taxable`, :guilabel:`Not Taxable`, :guilabel:`Taxable With Rate=0.00`, :guilabel:`Exempt`,
    :guilabel:`Suspended`
```
- Minor detail, but each element should be a separate guilabel :)
- Name the fields / options as they appear (e.g., Taxable With Rate=0.00)
- Also, it could be nice to add some more info (i.e., something like the toolip)

<img width="419" height="161" alt="Image" src="https://github.com/user-attachments/assets/791e1cf7-e527-4d2d-b90a-7ccf5d1ffa54" />

Author: (I just copied/pasted this section when I reviewed the structure)
You're right, I updated it with the details from the tooltip, because it was not useful to just have the different options ;)

>    - :guilabel:`CSLL Taxable` if the company is subject to CSLL or not

Reviewer: ```suggestion
   - :guilabel:`CSLL Taxable`: if the company is subject to CSLL or not
```

.. tip::
   If it is a simplified regime, the ICMS rate must be configured. To do so, go to
   :menuselection:`Accounting --> Configuration --> Settings`, scroll down to the :guilabel:`Taxes`
   section, and set the :guilabel:`Sales Tax` and :guilabel:`Purchase Tax` fields in the
   :guilabel:`Default Taxes` section.

The same configuration applies to the relevant :doc:`contact <../../essentials/contacts>` form when
using the AvaTax integration.

.. note::
   Select the :guilabel:`Company` option for a contact with a tax ID (CNPJ), or check

>    :guilabel:`Individual` for a contact with a CPF

Reviewer: ```suggestion
   :guilabel:`Individual` for a contact with a CPF.
```

.. _localizations/brazil/contacts-fiscal-reform:

> Fiscal reform

Reviewer: Could the info under this heading be merged with the previous section?

-------------

Following the 2026 Tax Reform, configure the :guilabel:`Fiscal Information` within the

> :guilabel:`Sales and Purchase` tab of the contacts form:

Reviewer: ```suggestion
:guilabel:`Sales & Purchase` tab of the contacts form:
```

> - :guilabel:`Tax Regime`: For :guilabel:`simplified` tax regime (:guilabel:`Simples Nacional`), set

Reviewer: ```suggestion
- :guilabel:`Tax Regime`: For the :guilabel:`simplified` tax regime (:guilabel:`Simples Nacional`), set
```

     (IAPs) <../../essentials/in_app_purchase>` to compute taxes and handle electronic documents
     (e.g., :ref:`NF-e <localizations/brazil/e-invoice-goods>`, :ref:`NFS-e
     <localizations/brazil/e-invoice-services>`). Each action consumes credits from the `IAP credit
     balance <https://iap.odoo.com/iap/in-app-services/819>`_. On creation, new databases receive
     500 free credits.

To compute the goods and services tax and process electronic invoices, the following configurations
are needed:

- :ref:`Company <localizations/brazil/company-and-contacts>`
- :ref:`Contacts <localizations/brazil/company-and-contacts>`
- :ref:`AvaTax configuration<localizations/brazil/avatax-credentials>`.
- :ref:`Fiscal reform <localizations/brazil/avatax-tax-reform>`
- :ref:`A1 digital certificate <localizations/brazil/certificate-upload>`
- :ref:`Tax mapping <localizations/brazil/fiscal-positions>`
- :ref:`Products <localizations/brazil/avatax-products>`

.. _localizations/brazil/avatax-configuration:

Configuration
-------------

.. _localizations/brazil/avatax-credentials:

Credentials
~~~~~~~~~~~

> :ref:`Activate AvaTax in Odoo <avatax/credentials>` and, in the :guilabel:`AvaTax Brazil` section,

Reviewer: There is no "AvaTax" option to activate after installing the AvaTax Brazil module

   sandbox or production database, use a real email address, as it is needed to :ref:`connect to
   Avalara <avatax/prerequisites>` and set up the certificates, whether to test or use it on
   production.

   There are two different Brazilian Avalara Portals:

   - One for testing: https://portal.sandbox.avalarabrasil.com.br/
   - One for production: https://portal.avalarabrasil.com.br/

   When the account is created from Odoo, select the right environment. Moreover, the email used to
   open the account cannot be used to open another account. Save the :guilabel:`API ID` and
   :guilabel:`API Key` when the account is created from Odoo.

   .. image:: brazil/transfer-api-credentials.png
      :alt: Transfer API Credentials.

After the account is created from Odoo, go to the Avalara Portal to set up the password:

#. Access the `Avalara portal <https://portal.avalarabrasil.com.br/Login>`_.
#. Click :guilabel:`Meu primeiro acesso`.
#. Add the email address used in Odoo to create the Avalara/AvaTax account, and click
   :guilabel:`Solicitar Senha`.
#. An email will then be received with a token and a link to create a password. Click this link

>    and copy-paste the token to allocate the desired password.

Reviewer: ```suggestion
   and copy-paste the token to set the desired password.
```

.. tip::
   If you use AvaTax in Odoo for tax computation **only**, setting a password or accessing the
   Avalara portal is unnecessary. However, to use the electronic invoice service, access to
   AvaTax is needed, and the :ref:`certificate must be uploaded
   <localizations/brazil/certificate-upload>`.

.. note::
   |API| credentials can be transferred. This option should be used only when an account has already
   been created in another Odoo instance and must be reused.

.. _localizations/brazil/avatax-tax-reform:

Fiscal reform
~~~~~~~~~~~~~

> To enable the sending of the new attributes of the 2026 Tax reform, follow these steps:

Reviewer: ```suggestion
To enable the sending of the new attributes of the 2026 tax reform, follow these steps:
```

#. Activate :ref:`developer mode <developer-mode>`.
#. Go to :menuselection:`Accounting --> Configuration --> Settings` and scroll to the
   :guilabel:`Taxes` section.

> #. Under :guilabel:`AvaTax Brazil`, enable :guilabel:`Enable Brazilian fiscal reform`.

Reviewer: I didn't fully configure it (i.e., I didn't create an AvaTax account) so I assume this option only appears once an account is set up?

Author: The option is available even before the AvaTax account is set up. And actually, developer mode isn't needed.

<https://portal.avalarabrasil.com.br/Login>`_.

The certificate will be synchronized with Odoo as long as the external identifier number in the
AvaTax portal matches, without special characters, with the CNPJ number, and the identification
number (CNPJ) in Odoo matches the CNPJ in AvaTax.

.. important::
   Some cities require the certificate to be linked within the City Portal system before issuing
   NFS-e from Odoo.

   If an error message from the city that says :guilabel:`Your certificate is not linked
   to the user` is received, this process needs to be done in the city portal.

.. _localizations/brazil/fiscal-positions:

Fiscal positions
~~~~~~~~~~~~~~~~

To set up the :guilabel:`Automatic Tax Mapping (Avalara Brazil)` :ref:`fiscal position
<avatax/fiscal_positions>`, enable the :guilabel:`Detect Automatically` and :guilabel:`Use AvaTax
Brazil API` options.

.. seealso::
   :doc:`Fiscal positions <../accounting/taxes/fiscal_positions>`

.. _localizations/brazil/avatax-products:

Products
~~~~~~~~

To use the AvaTax integration on sales orders and invoices, enter the following information in the

> :guilabel:`Sales` tab of the product form under the :guilabel:`Brazil Accounting` section, based on

Reviewer: I don't see that, for some reason it's all under the General Information tab

<img width="907" height="527" alt="Image" src="https://github.com/user-attachments/assets/ad43781b-e26d-45e2-852d-a1577e6958d8" />

   - :ref:`Invoice cancellation <localizations/brazil/invoice-cancellation>`
   - :ref:`Sales refund via credit note <localizations/brazil/credit-notes>`
   - :ref:`Sales complementary invoice via debit note <localizations/brazil/debit-notes>`
   - :ref:`Invoice number range invalidation <localizations/brazil/invoice-number-invalidation>`
   - Other tax validations.

.. note::
   If taxes are calculated in the **Sales** app, and the invoice is later issued in the
   **Accounting** app, the calculation happens twice, consuming two credits.

.. example::
   | **Sales order confirmed**
   | :icon:`fa-arrow-down` 1 credit (tax calculation)
   | **Invoice created**
   | :icon:`fa-arrow-down` 1 credit (tax calculation)
   | **Invoice confirmed and submitted**
   | :icon:`fa-arrow-down` 1 credit (tax calculation) + 1 credit (submit invoice)
   | **Total: 4 credits**

- :guilabel:`CEST Code`: tax classification code identifying goods and products subject to tax
  substitution under ICMS regulations, and helps determine the applicable tax treatment and
  procedures for specific items. The product's applicability to this requirement can be verified at

>   https://www.codigocest.com.br/.

Reviewer: ```suggestion
  procedures for specific items. The product's applicability to this requirement can be verified on the `Código CEST website <https://www.codigocest.com.br>`_.
```

- :guilabel:`Mercosul NCM Code`: Mercosur Common Nomenclature Product Code
- :guilabel:`Source of Origin`: origin of the product, which can be foreign or domestic, among other
  possible options, depending on the specific use case
- :guilabel:`SPED Fiscal Product Type`: fiscal product type according to the SPED list table
- :guilabel:`Purpose of Use`: intended purpose of use for this product

.. note::
   Odoo automatically creates three products to be used for transportation costs associated with
   sales. These are named :guilabel:`Freight`, :guilabel:`Insurance`, and :guilabel:`Other Costs`
   and are already configured. If more need to be created, duplicate and use the same configuration:

>    - :guilabel:`Product Type` :guilabel:`Service`

Reviewer: ```suggestion
   - :guilabel:`Product Type`: :guilabel:`Service`
```

.. seealso::
   :doc:`Fiscal positions (tax and account mapping) <../accounting/taxes/fiscal_positions>`

.. _localizations/brazil/accounting:

Accounting
==========

.. _localizations/brazil/electronic-documents:

Electronic documents
--------------------

.. _localizations/brazil/journals:

Configuration
~~~~~~~~~~~~~

A *series* number is linked to a sequence number range for electronic invoices. To configure the

> series number on a sales journal, go to :menuselection:`Accounting --> Configuration --> Journals`

Reviewer: ```suggestion
series number on a sales journal, go to :menuselection:`Accounting --> Configuration --> Journals` and select the journal
```

and set it in the :guilabel:`Series` field. If more than one series is needed, a new sales journal
must be created, and a new series number must be assigned for each series.

Enable the :guilabel:`Use Documents?` option as the :guilabel:`Series` field will only be displayed

> if the :guilabel:`Use Documents?` option is selected on the journal.

Reviewer: I would mention that first, as it's the first step before you can configure the series field

When issuing electronic and non-electronic invoices, the :guilabel:`Type` field selects the document
type used when creating the invoice.

.. note::

>    When creating the journal, ensure the :guilabel:`Dedicated Credit Note Sequence` field in the

Reviewer: Saying creating kind of implies you are creating a new journal, but you could be configuring an existing one, no?

Author: In this case, it's about the journal that has been created. The "Dedicated Credit Note Sequence" option is already unchecked in the default sales journal.

   :guilabel:`Accounting Information` section is unchecked, as in Brazil, sequences between
   invoices, credit notes, and debit notes are shared per series number, which means per journal.

.. _localizations/brazil/customer-invoices:

Customer invoices
~~~~~~~~~~~~~~~~~

To process an electronic invoice for goods (NF-e) or services (NFS-e), the invoice must be confirmed
and taxes must be computed by Avalara. The following fields must be filled out:

- :guilabel:`Customer`, with all customer information

> - :guilabel:`Payment Method: Brazil`: Specify the expected payment method.

Reviewer: I don't see the Payment Method: Brazil - but isn't that field (Payment Method) displayed when you register the payment? I guess it's because I only partially configured the DB

Author: I can't test it either, but it's surely a Brazil-specific field different from the Payment method one.

they are not currently used **and** will not be used in the future. To do so, go to
:menuselection:`Accounting --> Configuration --> Journals`, open the journal, click the
:icon:`fa-cog` (gear) icon, and select :guilabel:`Invalidate Number Range (BR)`. On the
:guilabel:`Invalidate Number Range (BR)` wizard, add the :guilabel:`Initial Number` and
:guilabel:`End Number` of the range that should be cancelled, and enter an invalidation
:guilabel:`Reason`.

.. note::
   - Invoice number range invalidation is only available for electronic invoices for goods (NF-e).
   - The journal's chatter records the log of the cancelled numbers, along with the XML file.

.. _localizations/brazil/vendor-bills:

Vendor bills
------------

When receiving an invoice from a supplier, encode the bill in Odoo by adding all the commercial
information and the same Brazilian-specific information recorded on the :ref:`customer invoices
<localizations/brazil/electronic-documents>`.

These Brazilian-specific fields are:

- :guilabel:`Payment Method: Brazil`: Specify the expected payment method.
- :guilabel:`Document Type`: used by the vendor
- :guilabel:`Document Number`: the invoice number from the supplier

> - :guilabel:`Freight Model`: **NF-e specific** how goods are planned to be transported - domestic

Reviewer: ```suggestion
- :guilabel:`Freight Model` (NF-e specific): how goods are planned to be transported - domestic
```

> - :guilabel:`Transporter Brazil`: **NF-e specific** who is doing the transportation.

Reviewer: ```suggestion
- :guilabel:`Transporter Brazil` (NF-e specific): who is doing the transportation.
```
